### PR TITLE
CUMULUS-1557: Update CI configuration to work with main TF branch

### PR DESCRIPTION
**Pull request by Jkovarik**

### CODE REVIEW

File: bamboo/set-bamboo-env-variables.sh [hunk 1]
```diff
@@ -101,27 +102,28 @@ fi
...
   export AWS_ACCESS_KEY_ID=$bamboo_SECRET_NONPROD_AWS_ACCESS_KEY_ID
   export AWS_SECRET_ACCESS_KEY=$bamboo_SECRET_NONPROD_AWS_SECRET_ACCESS_KEY
 fi
 
 export PR_BRANCH=master
-if [[  $DEPLOYMENT =~ '-tf' ]]; then
+if [[  $KES_DEPLOYMENT != true ]]; then
   echo "Setting GIT_PR target branch to 'terraform'"
   export PR_BRANCH=terraform
 fi
```
Comment (Jkovarik) [inline bamboo/set-bamboo-env-variables.sh]: This needs to go as soon as we put the PR in for master

File: packages/integration-tests/index.js [hunk 2]
```diff
@@ -539,7 +539,7 @@ async function addRulesWithPostfix(config, dataDirectory, overrides, postfix) {
 
       rule = Object.assign(rule, overrides);
       const ruleTemplate = Handlebars.compile(JSON.stringify(rule));
-      const templatedRule = JSON.parse(ruleTemplate(config));
+      const templatedRule = JSON.parse(ruleTemplate(Object.assign({}, config, process.env)));
```
Comment (markdboyd) [inline packages/integration-tests/index.js]: I'm not sure I understand why we are merging `process.env` onto this object?

File: bamboo/set-bamboo-env-variables.sh (hunk 1 above)
Comment (markdboyd) [reply bamboo/set-bamboo-env-variables.sh]: Or not if we're continuing to support Kes deployments for a while?

File: example/.gitignore [hunk 3]
```diff
@@ -20,10 +20,9 @@ test_app/
 scripts/test_output/
```
[the remark this answers is not in the document]
Comment (nemreid) [reply example/.gitignore]: Already is: `example/.gitignore:10: *.env`

File: bamboo/bootstrap-tf-deployment.sh [hunk 4]
```diff
@@ -59,25 +64,25 @@ echo "terraform {
...
 # Deploy cumulus-tf via terraform
 echo "Deploying Cumulus example to $DEPLOYMENT"
-../terraform plan \
-  -out=terraform.tfplan \
+../terraform apply \
+  -auto-approve \
   -input=false \
   -var-file="../deployments/sandbox.tfvars" \
   -var-file="../deployments/$DEPLOYMENT.tfvars" \
+  -var "cumulus_message_adapter_lambda_layer_arn=arn:aws:lambda:us-east-1:$AWS_ACCOUNT_ID:layer:Cumulus_Message_Adapter:9" \
```
[the remark this answers is not in the document]
Comment (nemreid) [reply bamboo/bootstrap-tf-deployment.sh]: You'd still have to update the environment variable. I had similar reasoning with the `.tfversion` file, but decided that we version lock all the rest of our dependencies in the codebase, it makes sense to lock this one too. Especially if new layers with breaking changes come thru in the future.

[the remark this answers is not in the document]
Comment (nemreid) [reply bamboo/bootstrap-tf-deployment.sh]: e.g. avoid breaking backport branches that may still need to be supported.

File: packages/integration-tests/index.js (hunk 2 above)
Comment (nemreid) [reply packages/integration-tests/index.js]: There are some places in the specs where it relied on `config.awsAccountId` and `config.awsRegion` being defined in the test configs, especially template files. It didn't really make sense to recreate defining and committing those as they are in `app/config.yml` when they are available in the environment in CI, so instead they were moved into the `example/.env` file, but the test helpers and handlebars still expect them to be in that config object.

Comment (nemreid) [reply packages/integration-tests/index.js]: b70dd0051 👍

**Pull request merged**